File: puzzles/ruby/ascii-art.rb
@width = gets.to_i # the width of a letter represented in ASCII art
@height = gets.to_i # the height of a letter represented in ASCII art.
@text = gets.chomp.upcase

asciiArt = Array.new # the characters A to Z and ? represented in ASCII art.

@height.times do
  row = gets.chomp
  asciiArt.push(row)
end

# Prints the letter for every height level
asciiArt.each do |row|
  @text.each_char { |c|
    ascii = c.ord
    position = ascii - 'A'.ord
    unless position.between?(0, 25)
      position = 26
    end
    column = position * @width
    for i in column..(column + @width - 1) do
      print row[i]
    end
  }
  print "\n"
end

File: puzzles/js/ascii-art.js
'use strict';

const ALPHABET_SIZE = 27;
const width = parseInt(readline());
const height = parseInt(readline());
const text = readline().toUpperCase();
const totalWidth = width * ALPHABET_SIZE;
let asciiArt = [];

// Fill the ASCII ART array
for (let i = 0; i < height; ++i)
  asciiArt.push(readline());

// Print the letter for each height level
asciiArt.forEach((row) => {
  let output = "";
  for (let i = 0; i < text.length; ++i) {
    let asciiCode = text[i].charCodeAt(0);
    let position = asciiCode - 'A'.charCodeAt(0);
    if (position < 0 || position > 25)
      position = 26;
    let column = position * width;
    for (let i = column; i < (column + width); ++i)
      output += row[i];
  } // for
  print(output);
}); // forEach()
